File: dotnet_3/cs/rest/SCS/Imp/Calc.cs
using System;

namespace SCS.Imp
{
    public class Calc
    {
        public static string Subject(string op, double arg1 , double arg2 )
        {
            op = op.ToLower();
            
            double result;
            if (op == "pi")
                result = Math.PI;
            else if (op == "e")
                result = Math.E;
            else if (op == "sqrt")
                result = Math.Sqrt(arg1);
            else if (op == "log")
                result = Math.Log(arg1);
            else if (op == "sine")
                result = Math.Sin(arg1);
            else if (op == "cosine")
                result = Math.Cos(arg1);
            else if (op == "tangent")
                result = Math.Tan(arg1);
            else if (op == "plus")
                result = arg1 + arg2;
            else if (op == "subtract")
                result = arg1 - arg2;
            else if (op == "multiply")
                result = arg1 * arg2;
            else if (op == "divide")
                result = arg1 / arg2;
            else
                result = 0.0;

            return "" + result;
        }
    }
}

File: jdk_8_maven/cs/graphql/graphql-scs/src/main/kotlin/org/graphqlscs/type/Calc.kt

package org.graphqlscs.type

import org.springframework.stereotype.Component
import java.util.*

@Component
class Calc {

    fun subject(op: String, arg1: Double, arg2: Double): String {
        var op = op
        op = op.lowercase(Locale.getDefault())
        var result = 0.0
        if ("pi" == op) { //CONSTANT OPERATOR
            result = Math.PI
        } else if ("e" == op) {
            result = Math.E
        } //UNARY OPERATOR
        else if ("sqrt" == op) {
            result = Math.sqrt(arg1)
        } else if ("log" == op) {
            result = Math.log(arg1)
        } else if ("sine" == op) {
            result = Math.sin(arg1)
        } else if ("cosine" == op) {
            result = Math.cos(arg1)
        } else if ("tangent" == op) {
            result = Math.tan(arg1)
        } //BINARY OPERATOR
        else if ("plus" == op) {
            result = arg1 + arg2
        } else if ("subtract" == op) {
            result = arg1 - arg2
        } else if ("multiply" == op) {
            result = arg1 * arg2
        } else if ("divide" == op) {
            result = arg1 / arg2
        }
        return "" + result
    }
}